src/helpers.js

```diff
@@ -34,4 +34,50 @@ function getSetImmediate() {
   }
 }
 
-export {getDocument, newMutationObserver, getSetImmediate}
+let originalSetTimeout = global.setTimeout
+
+if (typeof window !== 'undefined') {
+  originalSetTimeout = window.setTimeout
+}
+
+let originalClearTimeout = global.clearTimeout
+
+if (typeof window !== 'undefined') {
+  originalClearTimeout = window.clearTimeout
+}
+
+/*
+ * Return the original setTimeout function so that all functions work as expected by a user
+ * if they utilize dom-testing-library in a test where jest.fakeTimers() is used.
+ *
+ * global.useFakeTimers is used by us to make sure that we can still utilize fakeTimers in our tests.
+ *
+ * see: https://github.com/testing-library/dom-testing-library/issues/300
+ */
+function getSetTimeout() {
+  if (global.useFakeTimers) {
+    // eslint-disable-next-line no-negated-condition
+    return typeof window !== 'undefined' ? window.setTimeout : global.setTimeout
```

Comment:
You could avoid that eslint issue with:

```suggestion
    return typeof window === 'undefined' ? global.setTimeout : window.setTimeout
```

Reply:
Same as below